jsdom: window.document.defaultView !== window

### Describe the bug

I'm migrating tests from Jest to Vitest (both using jsdom) and stumbled upon this subtle difference between Jest and Vitest:

```js
it("works", () => {
  console.log(
    "document.defaultView === window",
    document.defaultView === window
  );

  const element = document.createElement("div");
  document.body.appendChild(element);
  console.log(
    "element.ownerDocument.defaultView === window",
    element.ownerDocument.defaultView === window
  );
});
```

See https://github.com/molily/vitest-test2/blob/main/first.test.js

Output in vitest:

```js
document.defaultView === window false
element.ownerDocument.defaultView === window false
```

Output in jest:

```js
document.defaultView === window true 
element.ownerDocument.defaultView === window true
```

In Vitest, `window` and `document.defaultView` are two different Window instances. I would expect both to point to the same object.

---

Why is this relevant? In my tests, I mock objects on `window` and expect that it affects all elements in the document.

My concrete test uses `@testing-library/dom` to fire an event on an element: https://github.com/molily/vitest-test2/blob/main/dom.test.js

 I'm mocking `window.PointerEvent` since jsdom does not support it:

```js
window.PointerEvent = PointerEventFake;
```

Then I use `fireEvent` from `@testing-library/dom` to dispatch an event. `fireEvent` uses `element.ownerDocument.defaultView` to get the associated Window object [(source 1)](https://github.com/testing-library/dom-testing-library/blob/main/src/helpers.ts#L27-L33). It then creates an event using the constructor on this obtained Window instance [(source 2)](https://github.com/testing-library/dom-testing-library/blob/main/src/events.js#L49-L50). In Vitest, `defaultView` yields a different object than `window`, so `@testing-library/dom` does not find my `PointerEvent` mock.

A workaround would be to assign the mock on `document.defaultView`:

```js
document.defaultView.PointerEvent = PointerEvent;
```

But this is obscure and most developers don't know what `defaultView` is (they shouldn't for this use case IMHO).

Is there a particular reason why `[DOM node].defaultView` isn't identical to `window`? Or am I doing anything wrong?

### Reproduction

Example repo: https://github.com/molily/vitest-test2

```
npm run vitest
```

yields

```
stdout | first.test.js > works
document.defaultView === window false
element.ownerDocument.defaultView === window false

 ✓ first.test.js (1)
 ❯ dom.test.js (1)
   × works

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  dom.test.js > works
AssertionError: expected false to be true // Object.is equality
 ❯ dom.test.js:37:18
     35|   document.body.appendChild(element);
     36|   fireEvent.pointerDown(element, { pointerType: "mouse" });
     37|   expect(called).toBe(true);
       |                  ^
     38| });
     39|

  - Expected   "true"
  + Received   "false"

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed | 1 passed (2)
      Tests  1 failed | 1 passed (2)
   Start at  10:07:22
   Duration  691ms (transform 218ms, setup 0ms, collect 61ms, tests 11ms)
```

whereas

```
npm run jest
```

yields

```
 PASS  ./dom.test.js
  ● Console

    console.log
      mouse click!
      at HTMLDivElement.log (dom.test.js:31:15)

 PASS  ./first.test.js
  ● Console

    console.log
      document.defaultView === window true
      at Object.log (first.test.js:2:11)

    console.log
      element.ownerDocument.defaultView === window true
      at Object.log (first.test.js:9:11)


Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        0.482 s, estimated 1 s
Ran all test suites.
```

### System Info

```shell
Node: 18.13.0
vitest: 0.27.1
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vitest/src/integrations/env/utils.ts
+++ b/packages/vitest/src/integrations/env/utils.ts
@@ -72,11 +72,14 @@ export function populateGlobal(global: any, win: any, options: PopulateOptions =
   if (global.global)
     global.global = global
 
-  Object.defineProperty(global.document, 'defaultView', {
-    get: () => global,
-    enumerable: true,
-    configurable: true,
-  })
+  // rewrite defaultView to reference the same global context
+  if (global.document && global.document.defaultView) {
+    Object.defineProperty(global.document, 'defaultView', {
+      get: () => global,
+      enumerable: true,
+      configurable: true,
+    })
+  }
 
   skipKeys.forEach(k => keys.add(k))
 

--- a/packages/vitest/src/integrations/env/utils.ts
+++ b/packages/vitest/src/integrations/env/utils.ts
@@ -72,6 +72,12 @@ export function populateGlobal(global: any, win: any, options: PopulateOptions =
   if (global.global)
     global.global = global
 
+  Object.defineProperty(global.document, 'defaultView', {
+    get: () => global,
+    enumerable: true,
+    configurable: true,
+  })
+
   skipKeys.forEach(k => keys.add(k))
 
   return {

--- a/test/core/test/dom.test.ts
+++ b/test/core/test/dom.test.ts
@@ -130,10 +130,15 @@ it('can call global functions without window works as expected', async () => {
 })
 
 it('globals are the same', () => {
+  expect(window).toBe(globalThis)
+  expect(window).toBe(global)
   expect(window.globalThis).toBe(globalThis)
   expect(window.Blob).toBe(globalThis.Blob)
   expect(window.globalThis.Blob).toBe(globalThis.Blob)
   expect(Blob).toBe(globalThis.Blob)
+  expect(document.defaultView).toBe(window)
+  const el = document.createElement('div')
+  expect(el.ownerDocument.defaultView).toBe(globalThis)
 })
 
 it('can extend global class', () => {

--- a/test/core/test/happy-dom.test.ts
+++ b/test/core/test/happy-dom.test.ts
@@ -102,8 +102,14 @@ it('can call global functions without window works as expected', async () => {
 })
 
 it('globals are the same', () => {
+  expect(window).toBe(globalThis)
+  expect(window).toBe(global)
   expect(window.globalThis).toBe(globalThis)
   expect(window.Blob).toBe(globalThis.Blob)
   expect(window.globalThis.Blob).toBe(globalThis.Blob)
   expect(Blob).toBe(globalThis.Blob)
+  expect(document.defaultView).toBe(window)
+  expect(document.defaultView).toBe(globalThis)
+  const el = document.createElement('div')
+  expect(el.ownerDocument.defaultView).toBe(globalThis)
 })

